state-synchronizer/src/tests/fuzzing.rs

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    chunk_request::{GetChunkRequest, TargetType},
+    chunk_response::{GetChunkResponse, ResponseLedgerInfo},
+    coordinator::SyncCoordinator,
+    network::{StateSynchronizerMsg, StateSynchronizerSender},
+    tests::{
+        helpers::{MockExecutorProxy, SynchronizerEnvHelper},
+        mock_storage::MockStorage,
+    },
+};
+use channel::{libra_channel, message_queues::QueueStyle};
+use futures::channel::mpsc;
+use libra_config::{
+    config::{NodeConfig, PeerNetworkId, RoleType},
+    network_id::{NetworkId, NodeNetworkId},
+};
+use libra_proptest_helpers::ValueGenerator;
+use libra_types::{
+    ledger_info::LedgerInfoWithSignatures, transaction::TransactionListWithProof,
+    waypoint::Waypoint, PeerId,
+};
+use network::{
+    peer_manager::{ConnectionRequestSender, PeerManagerRequestSender},
+    protocols::network::NewNetworkSender,
+};
+use proptest::{
+    arbitrary::{any, Arbitrary},
+    option,
+    prelude::*,
+    strategy::Strategy,
+};
+use std::{
+    collections::HashMap,
+    num::NonZeroUsize,
+    sync::{Arc, RwLock},
+};
+
+#[test]
+fn test_state_sync_fuzzer() {
+    let mut gen = ValueGenerator::new();
+    let data = generate_corpus(&mut gen);
+    fuzzer(data);
+}
+
+pub fn generate_corpus(gen: &mut ValueGenerator) -> Vec<u8> {
+    let msg = gen.generate(state_sync_msg_strategy());
+    lcs::to_bytes(&msg).expect("[fuzzing] failed to generate corpus for state sync msg")
+}
+
+pub fn fuzzer(data: Vec<u8>) {
```

Comment:
if you change this to &[u8] it should work

Reply:
with &[u8] Rust complains about lifetime later when I pass it into the `block_on` call